### PR TITLE
[DuckPlayer] 6 - Init updates and Watch on YouTube

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/0/1201141132935289/1207777888338886/f
Tech Design URL:
CC:

Description:
- Updates DuckPlayer initialization requirements
- Renames Youtubplayernavhandler to DuckPlayernavhandler
- Adds DuckPlayer logging
- Opens video on Youtube when you tap "Watch in Youtube" button
- Opens video on Youtube player when you tap the "Youtube" logo in the player itself

**Steps to test this PR**:

⚠️ This PR requires a specific branch of CSS to work.  Drag a local version of [content-scope-scripts](https://github.com/duckduckgo/content-scope-scripts) package to the app, and point to branch: `pr-releases/pr-993`
 
- [x] Set DuckPlayer to "Always"
- [x] Go to youtube.com and open a video
- [x] Tap "Watch" in YouTube button and confirm it opens in YouTube
- [x] Go back to DuckPlayer
- [x] Tap the Youtube logo in the player and confirm it opens on YouTube

### CODE REVIEW

File: DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift [hunk 1]
```diff
@@ -94,6 +99,17 @@ final class YoutubePlayerNavigationHandler {
         performNavigation(duckPlayerRequest, responseHTML: html, webView: webView)
     }
     
+    // Re-enables DP if required
+    private func updateTemporaryState(_ url: URL?) {
```
Comment (afterxleep) [inline DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift:103]: DP can now be temporarily disabled for specific cases (Such as Watch in Youtube) scenarios.   The basic logic is that if the video you're about to watch is the same video you're already watching, it should open in Youtube and not Duckplayer.

File: DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift [hunk 2]
```diff
@@ -105,6 +121,43 @@ extension YoutubePlayerNavigationHandler: DuckNavigationHandling {
...
+            completion(.allow)
+            return
+        }
+        
+        
+        os_log("DP: Handling DuckPlayer Player Navigation for %s", log: .duckPlayerLog, type: .debug, navigationAction.request.url?.absoluteString ?? "")
+       
+        // Handle Open in Youtube Links
+        // duck://player/watchInYoutube?v=12345
+        if let url = navigationAction.request.url,
```
Comment (afterxleep) [inline DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift:142]: The "Watch in Youtube" button fires up this type of link.  This is the way Windows and Android take care of that.

File: DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift [hunk 3]
```diff
@@ -137,29 +194,42 @@ extension YoutubePlayerNavigationHandler: DuckNavigationHandling {
...
     }
     
     // Handle URL changes not triggered via Omnibar
     // such as changes triggered via JS
     @MainActor
     func handleURLChange(url: URL?, webView: WKWebView) {
         
+        // If trying to load the same video while DP is visible
+        // Just open it in Youtube
+        if let url = url,
```
Comment (afterxleep) [inline DuckDuckGo/DuckPlayer/YouTubePlayerNavigationHandler.swift:213]: DP can now be temporarily disabled for specific cases (Such as Watch in Youtube) scenarios. The basic logic is that if the video you're about to watch is the same video you're already watching, it should open in Youtube and not Duckplayer.

File: DuckDuckGo.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -2488,7 +2488,8 @@
 		D62EC3C12C248AF800FC9D04 /* DuckNavigationHandling.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DuckNavigationHandling.swift; sourceTree = "<group>"; };
 		D63657182A7BAE7C001AF19D /* EmailManagerRequestDelegate.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = EmailManagerRequestDelegate.swift; sourceTree = "<group>"; };
 		D63677F42BBDB1C300605BA5 /* DaxLogoNavbarTitle.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = DaxLogoNavbarTitle.swift; sourceTree = "<group>"; };
-		D63FF8892C1B21C2006DE24D /* YouTubePlayerNavigationHandler.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = YouTubePlayerNavigationHandler.swift; sourceTree = "<group>"; };
+		D636DDF02C468BA7003696BD /* content-scope-scripts */ = {isa = PBXFileReference; lastKnownFileType = wrapper; name = "content-scope-scripts"; path = "../content-scope-scripts"; sourceTree = "<group>"; };
```
Comment (Bunn) [inline DuckDuckGo.xcodeproj/project.pbxproj]: should probably remove this before merging :)

File: DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift [hunk 5]
```diff
@@ -137,38 +158,70 @@ extension YoutubePlayerNavigationHandler: DuckNavigationHandling {
...
+                newURL = URL.youtube(videoID, timestamp: timestamp)
+            } else {
+                os_log("DP: Duckplayer is NOT disabled.  Opening DuckPlayer", log: .duckPlayerLog, type: .debug)
+            }
+            
+            // Load the URL
             webView.load(URLRequest(url: newURL))
+            
+            // Add a delay before resetting to allow the Webview to properly render
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```
Comment (Bunn) [inline DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift:201]: Isn't there a webview callback to listen instead of adding delays?

**Review by Bunn — APPROVED** [review]
LGTM, some small comments

File: DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift (hunk 5 above)
Comment (afterxleep) [reply DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift:201]: This is more about letting the webView start the loading process, not rendering the page.  The only callback is the WKNavigation delegate, but we don't want anything to do with that here.   I'll update the comment so is more clear :)

**Pull request merged**